src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -19,15 +19,29 @@
 class DeprecationErrorHandler
 {
     const MODE_WEAK = 'weak';
-    const MODE_WEAK_VERBOSE = 'weak-verbose';
 
     private static $isRegistered = false;
 
-    public static function register($mode = false)
+    /**
+     * Registers and configures the deprecation handler.
+     *
+     * The following reporting modes are supported:
+     * - use "weak" to hide the deprecation report but keep a global count;
+     * - use "/some-regexp/" to stop the test suite whenever a deprecation
+     *   message matches the given regular expression;
+     * - use a number to define the upper bound of allowed deprecations,
+     *   making the test suite fail whenever more notices are trigerred.
+     *
+     * @param scalar $mode The reporting mode. Defaults to not allowing any deprecations.
```

Comment:
Is `scalar` supported by common IDEs or should we use `int|string|bool` instead?